src/app/pages/send/send-crypto-asset-form/stx/use-stx-broadcast-transaction.tsx

```diff
@@ -0,0 +1,73 @@
+import { useCallback } from 'react';
+import toast from 'react-hot-toast';
+import { useNavigate } from 'react-router-dom';
+
+import { StacksTransaction, deserializeTransaction } from '@stacks/transactions';
+
+import { logger } from '@shared/logger';
+import { RouteUrls } from '@shared/route-urls';
+
+import { useHomeTabs } from '@app/common/hooks/use-home-tabs';
+import { LoadingKeys } from '@app/common/hooks/use-loading';
+import { useSubmitTransactionCallback } from '@app/common/hooks/use-submit-stx-transaction';
+import { useSignTransactionSoftwareWallet } from '@app/store/transactions/transaction.hooks';
+
+export function useStxBroadcastTransaction(unsignedTx: string) {
+  const { setActiveTabActivity } = useHomeTabs();
+  const signSoftwareWalletTx = useSignTransactionSoftwareWallet();
+  const navigate = useNavigate();
+
+  const handlePreviewClose = useCallback(() => {
+    navigate(RouteUrls.Home);
+    void setActiveTabActivity();
+  }, [navigate, setActiveTabActivity]);
+
+  const broadcastTransactionFn = useSubmitTransactionCallback({
+    loadingKey: LoadingKeys.CONFIRM_DRAWER,
+  });
+
+  const broadcastTransactionAction = useCallback(
+    async (signedTx: StacksTransaction) => {
+      if (!signedTx) {
+        logger.error('Cannot broadcast transaction, no tx in state');
+        toast.error('Unable to broadcast transaction');
+        return;
+      }
+      try {
+        await broadcastTransactionFn({
+          onClose() {
+            handlePreviewClose();
+          },
+          onError(e) {
+            handlePreviewClose();
+            navigate(RouteUrls.TransactionBroadcastError, { state: { message: e.message } });
+          },
+          replaceByFee: false,
+        })(signedTx);
+      } catch (e) {
+        handlePreviewClose();
+        navigate(RouteUrls.TransactionBroadcastError, {
+          state: { message: e instanceof Error ? e.message : 'unknown error' },
+        });
+      }
+    },
+    [broadcastTransactionFn, handlePreviewClose, navigate]
+  );
+
+  const broadcastTransaction = useCallback(
+    async (unsignedTx: StacksTransaction) => {
+      if (!unsignedTx) return;
+      const signedTx = signSoftwareWalletTx(unsignedTx);
+      if (!signedTx) return;
+      await broadcastTransactionAction(signedTx);
+    },
+    [broadcastTransactionAction, signSoftwareWalletTx]
+  );
+
+  const deserializedTransaction = deserializeTransaction(unsignedTx);
+
+  return {
+    stxDeserializedTransaction: deserializedTransaction,
+    stxBroadcastTransaction: broadcastTransaction,
+  };
```

Comment:
Maybe better a general `useMemo` that's returned rather than load of useCallsbacks? Gather this is existing code though